### PR TITLE
added modular arithmetic

### LINKED ISSUE #884: Modular Arithmetic in C++
<hr>

## Description 📜

I want to Create an Documentation explaining the concepts of Modular Arithmetic in C++ with some code examples too.

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

### PR DESCRIPTION
**Pull request by kaustubh0777**
<hr>

## Description 📜
Explained About Modular Arithmetic in C++ and its Properties ,Uses and Code Examples in C++. I have also mentioned some important algorithms based on Modular Arithmetic.
Fixes #884 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by kashishahuja2002 — COMMENTED** [review]
Good to go. @Suvraneel

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
Add link in the readme file.

File: C_CPP/modular_arithmetic.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+ Use->  Suppose we need to find the remainder after adding two large numbers in Programming then this property will be helpful as we can avoid the overflow and also the calculation will be fast enough to get the accurate result.
+
+ Example:-
+ <br>
+ <b>Ques-> (a+b)%m</b>
+
+ Code 
+ ----------------------------
+
```
Comment (Suvraneel) [inline C_CPP/modular_arithmetic.md:28]:
```suggestion
```c
```

File: C_CPP/modular_arithmetic.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+ Use-> If a and b both are large such that multiplying them is difficult .There we can use the concpet of modular multiplication.
+
+ Example:-
+ <br>
+ <b>Ques->(a*b)%m</b>
+
+ Code 
+ ----------------------------
+
+
```
Comment (Suvraneel) [inline C_CPP/modular_arithmetic.md:60]:
```c

**Comment by kashishahuja2002** [comment]
Where ever you have written code, just enclose them in three backticks ( ` ).

**Pull request closed without merging**